state/watcher/txnwatcher.go

```diff
@@ -167,6 +173,17 @@ func (w *TxnWatcher) Err() error {
 	return w.tomb.Err()
 }
 
+// Report is part of the watcher/runner Reporting interface, to expose runtime details of the watcher.
+func (w *TxnWatcher) Report() map[string]interface{} {
+	// TODO: (jam) do we need to synchronize with the loop?
```

Comment:
If we didn't care about sync-events-len or sync-events-cap we could just use atomic increment and atomic load for the values. We probably don't want to put atomic locks around every append to w.syncEvents.

Reply:
I think we care about sync-queue-cap, but in all the time I spent monitoring sync-queue-len it was always 0 by the time we came to monitor it. I think I could do the same thing as I did with HubWatcher and track the 'average' length from the last sync, and that would be useful.

request-queue-cap and request-queue-len were always 0 for me, because those represent watching a document that has changed since we loaded the document.  I actually never had 'request-queue-cap' anything but 0, which means we never generated an event based on watching a document that we had already witnessed a change to.
Its a lot of buffering for something that never happens...

Reply:
Ah, but that is a different watcher. for this one, I'll just use the same pattern of using a channel to coordinate with loop() rather than locking the various values.